performance/main.py

```diff
@@ -14,32 +14,135 @@
 #    under the License.
 
 import flask
-from flask_helpers import routing  # noqa
+import requests
+import os
+import json
 
-from performance import api
-from performance import config
 
-app = flask.Flask(__name__, static_folder=None)
-app.config.update(config.get_config()["flask"])
 
 
-for module in [api]:
-    for url_prefix, blueprint in module.get_blueprints():
-        app.register_blueprint(blueprint, url_prefix="/api%s" % url_prefix)
-
-
-app = routing.add_routing_map(app, html_uri=None, json_uri="/")
-
+app = flask.Flask(__name__)
+es_url = ''
+es_index_name = 'performance'
+es_index_type = 'key_metrics'
 
 @app.errorhandler(404)
 def not_found(error):
     return flask.jsonify({"error": "Not Found"}), 404
 
 
 def main():
+
+    etcd_host = os.environ['ETCD_HOST']
+    # For debug 
+    pretty_js = lambda x: json.dumps(x, sort_keys=True, indent=4)   
+
+    r = requests.get('http://' + etcd_host + ':2379/v2/keys/elasticsearch/endpoints/host')
+
+    es_host = r.json()['node']['value']
+
+    r = requests.get('http://' + etcd_host + ':2379/v2/keys/elasticsearch/endpoints/port')
+    es_port = r.json()['node']['value']
+
+    global es_url
+    es_url = 'http://' + es_host + ':' + es_port 
+    
     app.run(host=app.config.get("HOST", "0.0.0.0"),
             port=app.config.get("PORT", 5010))
 
 
+# var1/var2/var3
+
+#@app.route("/<region>/<metric>/<agg>")
+#def get_health(region, metric, agg):
+#
+#    request_body =  {
+#        'size': 0,
+#            'query': {
+#                'match': {
+#                    'region': region
+#                 }
+#            },
+#        'aggs': {
+#            'agg_name': {
+#                'filter' : { 'term': { 'metric': metric } },
+#                'aggs' : { 'avg_value' : { agg : { "field" : "value" } } }
+#            }
+#        }
+#    }
+#
+#
+#    r = requests.post(es_url + "/_search?&pretty", json=request_body )
+#
+#    return flask.jsonify([r.json()])
+#
+
+
+#api/v1/region/<name_of_region>/performance/week
+@app.route("/<region>/<metric>/<period>")
+def get_health(region, metric, period):
+
+    period_map = {
+      'day':   'now-1d',
+      'week':  'now-7d',
+      'month': 'now-1M'
+    }    
+    
+    interval_map = {
+      'day':  '15m',
+      'week': '100m',
+      'month': '3100m'
+    }
+    
+    request_body =  {
+        "size": 0,
+            "query": {
+                "bool": {
+                    "must": [
+                     {
+                         "range" : {
+                         "timestamp" : {
+                           "gte" : period_map[period],
+                           "lt" :  "now",
+                           "format" : "yyyy-MM-dd'T'H:m:s+00:00"
+                }
+            }
+        },
+        { "match": { "region": region }},
+        { "match": { "metric": metric }}
```

Comment:
Not sure that this is proper syntax. And we shoudn't do the quering per metric and region